content/develop/interact/search-and-query/advanced-concepts/vectors.md

```diff
@@ -9,369 +9,455 @@ categories:
+
+**Syntax**
+
+Vector search queries with filters follow this basic structure:
+
+```
+FT.SEARCH <index_name> <primary_filter_query>=>[...]
+```
 
-Currently, Redis doesn't support hybrid queries with multiple scoring options. Redis's approach combines pre-filtering documents against an index containing searchable fields (`TEXT`, `TAG`, `NUMERIC`, `GEO`, `GEOSHAPE`, and `VECTOR`) and generated embeddings, finding similarity between them according to the chosen algorithm (e.g., `KNN`).
+where `<primary_filter_query>` defines document selection and filtering.
 
-Pre-filter with a vector similarity KNN query has the form `<primary_filter_query>=>[<vector_similarity_query>]`. The first part defines document selection and filtering, and the second part is represented by query modifier attributes that define how similarity should be applied, in this case, KNN. Redis Stack has an internal mechanism for optimizing the computation of such queries. Two modes in which pre-filter queries are executed are:
+### How Filtering Works
 
-1. Batches mode - In this mode, a batch of the high-scoring documents from the vector index are retrieved. These documents are returned ONLY if `<primary_filter_query>` is satisfied. In other words, the document contains a similar vector and meets the filter criteria. The procedure terminates when `k` documents that pass the `<primary_filter_query>` are returned or after every vector in the index was obtained and processed.
+Redis uses internal algorithms to optimize the filtering computation for vector search. The runtime algorithm is determined by heuristics that aim to minimize query latency, based on several factors derived from the query and the index.
 
-The *batch size* is determined by a heuristics that is based on `k`, and the ratio between the expected number of documents in the index that pass the `<primary_filter_query>` and the vector index size.
-The goal of the heuristics is to minimize the total number of batches required to get the `k` results, while preserving a small batch size as possible.
-Note that the batch size may change *dynamically* in each iteration, based on the number of results that passed the filter in previous batches.
+**Batches mode**
 
-2. Ad-hoc brute-force mode - In general, this approach is preferable when the number of documents that pass the `<primary_filter_query>` part of the query is relatively small.
-   Here, the score of *every* vector which corresponds to a document that passes the filter is computed, and the top `k` results are selected and returned.
-   Note that the results of the KNN query will *always be accurate* in this mode, even if the underline vector index algorithm is an approximate one.
+- A batch of high-scoring documents from the vector index is retrieved. These documents are yielded ONLY if the `<primary_filter_query>` is satisfied. In other words, the document must contain a similar vector and meet the filter criteria.
```

Comment:
Consider comparing it to paginating over the vector index. I think it will give people the right context.